src/stores/__tests__/bots.spec.js

```diff
@@ -0,0 +1,83 @@
+import { setActivePinia, createPinia } from 'pinia'
+
+import {
+	getConversationBots,
+	disableBotForConversation,
+	enableBotForConversation,
+} from '../../services/botsService.js'
+import { generateOCSResponse } from '../../test-helpers.js'
+import { useBotsStore } from '../bots.js'
+
+jest.mock('../../services/botsService', () => ({
+	getConversationBots: jest.fn(),
+	disableBotForConversation: jest.fn(),
+	enableBotForConversation: jest.fn(),
+}))
+
+describe('botsStore', () => {
+	beforeEach(async () => {
+		setActivePinia(createPinia())
+	})
+
+	it('returns an empty array if bots are not available', async () => {
+		const botsStore = useBotsStore()
+		expect(botsStore.getConversationBots('empty-token')).toEqual([])
+	})
+
+	it('receives a response from server, stores and updates bots', async () => {
+		const token = 'TOKEN'
+		const botsStore = useBotsStore()
+
+		// First response
+		const oldBots = [
+			{ id: 1, name: 'Dummy bot 1', state: 0 },
+			{ id: 2, name: 'Dummy bot 2', state: 1 },
+		]
+		let response = generateOCSResponse({ payload: oldBots })
+		getConversationBots.mockResolvedValueOnce(response)
+		await botsStore.loadConversationBots(token)
+
+		expect(getConversationBots).toHaveBeenCalledWith(token)
+		expect(botsStore.getConversationBots(token)).toEqual(oldBots)
+
+		// New response with updated bots list
+		const newBot = { id: 3, name: 'Dummy bot 3', state: 2 }
+		response = generateOCSResponse({ payload: [...oldBots, newBot] })
+		getConversationBots.mockResolvedValueOnce(response)
+		await botsStore.loadConversationBots(token)
+
+		expect(getConversationBots).toHaveBeenCalledWith(token)
+		expect(botsStore.getConversationBots(token)).toEqual([...oldBots, newBot])
+	})
```

Comment:
For example, this test definitely is not one test. It testing failed and in the report we have
```
❌ receives a response from server, stores and updates bots
```

It is hard to understand, what was broken. Does it not load bots, or doesn't store them, or are there issues with updates?

It should be a load conversation bots test + update a bot test.

If many tests have exactly the same initial state and arrangement, it can be reused with `beforeEach`.